### PR TITLE
Harden CLI telemetry for agent-CLI audience

### PR DESCRIPTION
**Pull request by moona3k**
## Why

The CLI is being repositioned as the canonical Parakeet-on-Apple-Silicon surface for agent operators (OpenClaw / Hermes / scripted callers). For that audience the telemetry bar shifts: GUI norms (default-on, Settings toggle) aren't enough. Peer CLIs in this space honor `DO_NOT_TRACK`, auto-disable in CI, and document every byte of egress in the integration guide.

Collection itself was already minimal — random per-process session UUID, no paths / URLs / transcript content, server-side PII redaction. This PR closes the **disclosure-and-defaults gap** without changing payload, schema, exit codes, or stdout.

## What actually goes over the wire (unchanged — confirmed by audit)

```
   ┌─────────────────────────────────────────────────────────────────┐
   │  macparakeet-cli transcribe ~/Recordings/Standup-2026-04-28.m4a │
   └─────────────────────────────────────────────────────────────────┘
                                  │
                                  ▼
   ┌─────────────────────────────────────────────────────────────────┐
   │  STRIPPED at the typed event boundary (cannot reach the wire):  │
   │    • file path           (~/Recordings/Standup-2026-04-28.m4a)  │
   │    • filename            (Standup-2026-04-28.m4a)               │
   │    • YouTube URL / video ID                                     │
   │    • transcript text     (raw + clean)                          │
   │    • word count          (not in cli_operation, only GUI events)│
   │    • language flag value                                        │
   │    • custom database path                                       │
   │    • microphone name / device UID / hardware serial             │
   │    • LLM prompts and responses                                  │
   └─────────────────────────────────────────────────────────────────┘
                                  │
                                  ▼
   ┌─────────────────────────────────────────────────────────────────┐
   │  cli_operation event (one POST per invocation, batched/flushed) │
   │  ─────────────────────────────────────────────────────────────  │
   │    operation_id          random UUID, per-process               │
   │    workflow_id           random UUID, per-process               │
   │    parent_operation_id   nil for CLI                            │
   │    command               "transcribe"                           │
   │    subcommand            nil                                    │
   │    outcome               success | failure | cancelled | empty  │
   │    duration_seconds      12.5                                   │
   │    input_kind            audio | video | youtube | meeting      │
   │    output_format         "text" | "json"                        │
   │    json                  true | false                           │
   │    exit_code             0 | 1                                  │
   │    error_type            "URLError.timedOut" (classifier output)│
   │  + envelope                                                     │
   │    session               random UUID, per-process               │
   │    app_ver, os_ver, locale, chip                                │
   │    ts                    ISO 8601                               │
   │    event_id              UUID, idempotency key                  │
   └─────────────────────────────────────────────────────────────────┘
                                  │ HTTPS, 1s timeout
                                  ▼
   ┌─────────────────────────────────────────────────────────────────┐
   │  https://macparakeet.com/api/telemetry  (self-hosted CF Worker) │
   │    • allowlist validates event name (rejects unknown)           │
   │    • rate-limited 10 req/min/IP (IP not stored)                 │
   │    • country added from CF-IPCountry header                     │
   │    • PII redaction (paths, URLs, emails, API-key shapes)        │
   │    • inserted into D1 (SQLite), 90-day retention                │
   └─────────────────────────────────────────────────────────────────┘
```

The `cli_operation` payload is pinned by the new `CLIOperationPrivacyTests` — any future addition of a path-shaped, URL-shaped, or content-shaped prop fails CI at PR review time.

## New opt-out matrix

```
   Resolution order — first match wins, no fall-through:

   ┌─────────────────────────────────────────────┬─────────────────────┐
   │ MACPARAKEET_TELEMETRY=0/false/no/off        │ → force off         │
   │ MACPARAKEET_TELEMETRY=1/true/yes/on         │ → force on          │   ← NEW: defeats DNT/CI
   │ DO_NOT_TRACK=1                              │ → force off         │   ← NEW
   │ CI=true / GITHUB_ACTIONS=true / GITLAB_CI / │ → auto-disable      │   ← NEW
   │   BUILDKITE / CIRCLECI / TRAVIS /           │                     │
   │   JENKINS_URL / TF_BUILD / TEAMCITY_VERSION │                     │
   │ macparakeet-cli config set telemetry off    │ → persist off       │   ← NEW
   │ GUI Settings → Privacy → "Help improve…"    │ → persist off       │
   │ (default)                                   │ → on, honored       │
   └─────────────────────────────────────────────┴─────────────────────┘
```

`MACPARAKEET_TELEMETRY=1` is preserved as the explicit override for developers smoke-testing telemetry from a CI shell that has `DO_NOT_TRACK` or `CI` set globally.

## Changes

| Change | File | Why |
|---|---|---|
| Honor `DO_NOT_TRACK=1` | `CLITelemetry.swift` | Industry standard (Homebrew, GitLab, VS Code). Cheapest possible way to look like a peer to other agent CLIs. |
| Auto-disable in CI | `CLITelemetry.swift` | Prevents 1000-job agent runs flooding the endpoint with `cli_operation` events. Conservative — `CI=false` does not trigger. |
| `macparakeet-cli config get/set/list` | `Commands/ConfigCommand.swift` (new) | CLI-only users (brew install, no GUI) had no way to persist opt-out. Writes to the shared UserDefaults suite the GUI reads. |
| Surface telemetry in `transcribe --help` | `Commands/TranscribeCommand.swift` | Disclosure at the moment of consent. |
| "See the full event catalog" link | `Views/Settings/SettingsView.swift` | Lets users audit before deciding. Deep-links to `docs/telemetry.md` on GitHub. |
| Privacy bullet enumerates telemetry egress | `integrations/README.md` | Previously listed YouTube/LLM/Sparkle and silently omitted the `cli_operation` POST — the most agent-facing doc was misleading. |
| Env-var precedence ladder doc | `docs/telemetry.md` | Single source of truth for resolution order. |
| `Unreleased` changelog entry | `Sources/CLI/CHANGELOG.md` | Public CLI surface contract. |
| Env-var precedence tests | `Tests/CLITests/CLITelemetryTests.swift` (new) | 13 cases: every truthy/falsy synonym, CI detection edge cases, override-beats-DNT, override-beats-CI, force-off-still-force-off-in-CI. |
| Config persistence tests | `Tests/CLITests/ConfigCommandTests.swift` (new) | 11 cases against an isolated UserDefaults suite — never touches the real plist. |
| `cli_operation` prop allowlist | `Tests/MacParakeetTests/Services/CLIOperationPrivacyTests.swift` (new) | Pins schema; future attempts to add `file_path`, `youtube_url`, `transcript_excerpt`, etc. fail at PR review. |

## Behavior verification

`macparakeet-cli config --help`:
```
OVERVIEW: Read or write CLI/app configuration values.

Configuration is stored in the shared MacParakeet UserDefaults suite
(com.macparakeet.MacParakeet). The GUI reads the same suite, so values set here
apply to both surfaces.

Supported keys:
  telemetry   on|off   Anonymous, non-identifying usage analytics
                       (default: on). See docs/telemetry.md for the
                       full event catalog.

Per-process overrides (env vars, do not require `config set`):
  MACPARAKEET_TELEMETRY=0   Force-off for one invocation
  MACPARAKEET_TELEMETRY=1   Force-on for one invocation
  DO_NOT_TRACK=1            Force-off (industry-standard signal)
  CI=true                   Auto-disabled in CI environments
```

`macparakeet-cli transcribe --help` now opens with:
```
Telemetry: emits one privacy-safe `cli_operation` event per invocation
(command/outcome/duration/input_kind — never the path, URL, or transcript).
Disable with `MACPARAKEET_TELEMETRY=0`, `DO_NOT_TRACK=1`, the persistent
`macparakeet-cli config set telemetry off`, or the GUI Settings toggle.
Auto-disabled in CI (CI/GITHUB_ACTIONS/etc.). See docs/telemetry.md.
```

## What is intentionally NOT in this PR

- **No payload changes.** `cli_operation` schema is frozen by `CLIOperationPrivacyTests`; the PR was scoped to gating + disclosure.
- **No GUI behavior change** beyond the "Learn more" link. The Settings toggle text is unchanged.
- **No worker-side changes.** Allowlist, redaction, rate limiting, retention are unchanged.
- **No `telemetry_opted_out` event from CLI `config set`** — deferred. The GUI still emits it; CLI flips persist silently. Can revisit if we want exit-survey signal.
- **Crash reporter `reason` field hardening** — false positive in initial audit; `CrashReporter.objcExceptionHandler` already passes `reason` through `TelemetryErrorClassifier.errorDetail` which calls `sanitize()`. Verified at `Sources/MacParakeetCore/Services/CrashReporter.swift:227-229`.

## Test plan

- [x] `swift build --target CLI` — clean
- [x] `swift build --target MacParakeet` — clean
- [x] `swift test` — 1886 XCTest + 16 Swift Testing pass; no new failures
- [x] `swift run macparakeet-cli config --help` — help text renders correctly
- [x] `swift run macparakeet-cli transcribe --help` — telemetry note surfaces in help
- [ ] Smoke-test on macOS with GUI installed: GUI Settings "Learn more" link opens GitHub
- [ ] Smoke-test `macparakeet-cli config set telemetry off` then `config get telemetry` round-trip
- [ ] Smoke-test `MACPARAKEET_TELEMETRY=0 DO_NOT_TRACK=1 CI=true macparakeet-cli transcribe sample.wav` — confirm no telemetry POST (via `tcpdump` or proxy)

## ADRs

- **ADR-002** (local-first, opt-out telemetry, no audio/transcript content) — extended to the CLI/agent audience without amendment.
- **ADR-012** (self-hosted telemetry via Cloudflare Worker + D1) — payload and endpoint unchanged.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added CLI config command (get/set/list) to persist telemetry preference across CLI and GUI installs; accepts common boolean synonyms.
  * Telemetry now honors DO_NOT_TRACK and auto-disables in CI; explicit env override to force on/off is supported.
  * Settings UI adds a link to telemetry documentation.

* **Documentation**
  * Updated telemetry docs with full override hierarchy, emitted event allowlist, and opt-out options.

### CODE REVIEW

File: Sources/CLI/Commands/CLITelemetry.swift [hunk 1]
```diff
@@ -2,20 +2,87 @@ import Foundation
...
+    /// Pure decision function — testable without ProcessInfo or UserDefaults state.
+    /// `MACPARAKEET_TELEMETRY` wins outright. Then `DO_NOT_TRACK=1` (industry-standard,
+    /// honored by Homebrew, GitLab, VS Code, etc.). Then CI auto-disable so a 1000-job
+    /// agent run in GitHub Actions doesn't silently flood the endpoint with `cli_operation`
+    /// events.
+    static func decideOverride(env: [String: String]) -> EnvOverride {
+        if let raw = env["MACPARAKEET_TELEMETRY"]?.trimmingCharacters(in: .whitespacesAndNewlines).lowercased(),
+           !raw.isEmpty {
+            if ["0", "false", "no", "off"].contains(raw) { return .forceOff }
+            if ["1", "true", "yes", "on"].contains(raw) { return .forceOn }
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/CLI/Commands/CLITelemetry.swift:46]: The divergence is intentional and I'm keeping it. The env-var path stays minimal (`0/1/false/true/yes/no/on/off`) because env-var booleans are conventionally terse — `MACPARAKEET_TELEMETRY=enable` would read oddly. `ConfigCommand.parseBool` adds `enable`/`enabled`/`disable`/`disabled` because `macparakeet-cli config set telemetry enable` is the human-friendly shape that justifies a CLI subcommand existing in the first place. Unifying would either over-permit env vars or under-permit the config command. Happy to revisit if it bites later.

File: Sources/CLI/Commands/CLITelemetry.swift [hunk 2]
```diff
@@ -2,20 +2,87 @@ import Foundation
...
+            "CI",
+            "GITHUB_ACTIONS",
+            "GITLAB_CI",
+            "BUILDKITE",
+            "CIRCLECI",
+            "TRAVIS",
+            "JENKINS_URL",
+            "TF_BUILD",
+            "TEAMCITY_VERSION"
+        ]
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/CLI/Commands/CLITelemetry.swift]: Done in 5d754223 — hoisted to `private static let ciEnvVars` so it's not reallocated per call. Thanks.

File: Sources/CLI/Commands/ConfigCommand.swift [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            let parsed = try parseBool(value, key: key)
+            store.set(parsed, forKey: AppPreferences.telemetryEnabledKey)
+            return parsed ? "on" : "off"
+        default:
+            throw ConfigError.unknownKey(key)
+        }
+    }
+
+    static func parseBool(_ value: String, key: String) throws -> Bool {
+        let v = value.trimmingCharacters(in: .whitespaces).lowercased()
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/CLI/Commands/ConfigCommand.swift]: Already addressed in b06530f4 — `parseBool` now trims `.whitespacesAndNewlines`, and `testParseBoolTrimsNewlines` pins the behavior.

File: integrations/README.md [hunk 4]
```diff
@@ -266,10 +266,26 @@ macparakeet-cli prompts run "<prompt-name>" \
...
-  update checks (app, not CLI).
+- **Privacy:** STT and database access never touch the network. Network
+  egress paths are: YouTube downloads (yt-dlp); optional cloud LLM provider
+  calls (only when `prompts run --provider <cloud>` or `llm` against a hosted
+  provider); Sparkle update checks (app, not CLI); and a single privacy-safe
+  `cli_operation` event per `transcribe` invocation, posted to the
+  self-hosted endpoint at `https://macparakeet.com/api/telemetry`. The
+  telemetry event ships outcome / duration / `input_kind` only — never the
+  file path, URL, transcript, or any user content (random per-process
+  session UUID, no persistent identifier). Disable it any of four ways:
```
[the remark this answers is not in the document]
Comment (moona3k) [reply integrations/README.md]: Already addressed in b06530f4. Current text now reads: *"The telemetry event ships only allowlisted invocation metadata (`operation_id`, `workflow_id`, `parent_operation_id`, `command`, `subcommand`, `outcome`, `duration_seconds`, `input_kind`, `output_format`, `json`, `exit_code`, `error_type`) — never the file path, URL, transcript, language value, or any user content..."*

[the remark this answers is not in the document]
Comment (moona3k) [reply integrations/README.md]: Addressed in b06530f4 — same fix as the parallel CodeRabbit thread. README now lists the full allowlisted prop set explicitly instead of claiming "only outcome/duration/input_kind".

File: Sources/MacParakeet/Views/Settings/SettingsView.swift [hunk 5]
```diff
@@ -943,11 +943,27 @@ struct SettingsView: View {
...
+                )
+                Button {
+                    if let url = URL(string: "https://github.com/moona3k/macparakeet/blob/main/docs/telemetry.md") {
+                        NSWorkspace.shared.open(url)
+                    }
+                } label: {
+                    HStack(spacing: 4) {
+                        Text("See the full event catalog")
+                        Image(systemName: "arrow.up.right.square")
+                            .font(.caption)
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeet/Views/Settings/SettingsView.swift:960]: Done in 5d754223 — `.accessibilityHidden(true)` on the SF Symbol so VoiceOver reads only the button label + accessibilityHint, not the duplicated "square arrow up right" announcement.

**Comment by moona3k** [comment]
## Reviewer findings — triage and disposition

Total: 7 findings across gemini-code-assist, CodeRabbit, and Copilot. All triaged.

### ✅ Already addressed in earlier commits on this branch

| Finding | Reviewer | Commit |
|---|---|---|
| `parseBool` should use `.whitespacesAndNewlines` | gemini | `b06530f4` |
| `integrations/README.md` Privacy bullet under-reported `cli_operation` payload (claimed "outcome/duration/input_kind only") | CodeRabbit, Copilot | `b06530f4` |
| `CLIOperationPrivacyTests` should also assert *required* keys, not just absent forbidden ones | CodeRabbit | `b06530f4` |
| `transcribe --help` should reference a full URL, not `docs/telemetry.md` repo-relative | CodeRabbit | `b06530f4` |
| Unused `MacParakeetCore` import in `CLITelemetryTests.swift` | CodeRabbit | `b06530f4` |

### ✅ Addressed in this final pass — `5d754223`

| Finding | Reviewer | Action |
|---|---|---|
| `ciVars` reallocated per `isCIEnvironment` call | gemini | Hoisted to `private static let ciEnvVars` |
| Decorative `arrow.up.right.square` SF Symbol announced by VoiceOver | Copilot | `.accessibilityHidden(true)` on the icon; button label + `accessibilityHint` are the announced surface |

### ⏸ Won't fix — intentional design

| Finding | Reviewer | Rationale |
|---|---|---|
| Bool parsing logic in `decideOverride` duplicates `ConfigCommand.parseBool` and is "slightly inconsistent" | gemini | The divergence is deliberate. Env vars stay terse (`0/1/false/true/yes/no/on/off`) — `MACPARAKEET_TELEMETRY=enable` would read oddly. `config set telemetry` is permissive (also `enable`/`enabled`/`disable`/`disabled`) because that's the human-readable shape that justifies a CLI subcommand existing. Unifying would either over-permit env vars or under-permit the config command. |

### Final state
- 4 commits, all rich-formatted
- `swift test` → 1890 XCTest + 16 Swift Testing pass (verified locally on `5d754223`)
- `swift build --target CLI` and `--target MacParakeet` clean
- `config get bogus` and `config get bogus --json` both exit `2` per the documented contract; JSON envelope matches `CLIErrorEnvelope` schema

🤖 Generated with [Claude Code](https://claude.com/claude-code)
